src/formattedcode/output_cyclonedx.py

```diff
@@ -0,0 +1,659 @@
+    alg: str = attr.ib()
+    content: str = attr.ib()
+
+
+@attr.s
+class CycloneDxExternalRef:
+    url: str = attr.ib()
+    type: str = attr.ib(validator=attr.validators.in_(["vcs", "issue-tracker", "website",
+                                                       "advisories", "bom", "mailing-list", "social", "chat",
+                                                       "documentation", "support", "distribution", "license",
+                                                       "build-meta", "build-system", "other"]))
+    comment: str = attr.ib(default=None)
+    hashes: List[CycloneDxHashObject] = attr.ib(factory=list)
+
+
+class CycloneDxComponentType(str, Enum):
+    APPLICATION = "application"
+    FRAMEWORK = "framework"
+    LIBRARY = "library"
+    CONTAINER = "container"
+    OPERATING_SYSTEM = "operating-system"
+    DEVICE = "device"
+    FIRMWARE = "firmware"
+    FILE = "file"
+
+
+class CycloneDxComponentScope(str, Enum):
+    REQUIRED = "required"
+    OPTIONAL = "optional"
+    EXCLUDED = "excluded"
+
+
+@attr.s
+class CycloneDxComponent():
+    name: str = attr.ib()
+    version: str = attr.ib()
+    bom_ref: str = attr.ib(default=None)
+    group: str = attr.ib(default=None)
+    copyright: str = attr.ib(default=None)
+    author: str = attr.ib(default=None)
+    description: str = attr.ib(default=None)
+    purl: str = attr.ib(default=None)
+    hashes: List[CycloneDxHashObject] = attr.ib(factory=list)
+    licenses: List[CycloneDxLicenseEntry] = attr.ib(factory=list)
+    externalReferences: List[CycloneDxExternalRef] = attr.ib(factory=list)
+    type: CycloneDxComponentType = attr.ib(
+        default=CycloneDxComponentType.LIBRARY)
+    properties: List[CycloneDxAttribute] = attr.ib(factory=list)
+    scope: CycloneDxComponentScope = attr.ib(default=CycloneDxComponentScope.REQUIRED)
+
+
+@attr.s
+class CycloneDxDependency:
+    ref: str = attr.ib()
+    dependsOn: str = attr.ib(factory=list)
+
+
+@attr.s
+class CycloneDxBom():
+    bomFormat: str = attr.ib(init=False, default="CycloneDX")
+    specVersion: str = attr.ib(init=False, default="1.3")
+    serialNumber: str = attr.ib(
+        init=False, default="urn:uuid:" + str(uuid.uuid4()))
+    version: int = attr.ib(init=False, default=1)
+    metadata: CycloneDxMetadata = attr.ib(default=CycloneDxMetadata())
+    components: List[CycloneDxComponent] = attr.ib(factory=list)
+    dependencies: List[CycloneDxDependency] = attr.ib(factory=list)
+
+
+def get_tool_header(version: str) -> dict:
+    return {
+        "vendor": "nexB Inc.",
+        "name": "scancode-toolkit",
+        "version": version
+    }
+
+
+def get_external_ref_from_key(package: dict, key: str) -> CycloneDxExternalRef:
+    url = package.get(key)
+    type = url_type_mapping[key]
+    if url is not None and type is not None:
+        return CycloneDxExternalRef(url=url, type=type)
+    return None
+
+
+def get_license_entry_from_license_expression(license_expression: str) \
```

Comment:
nit: please avoid using backslash for line continuations. Wrapping black-style is best. 90'sih line length is OK too.
Also, could this be instead a class  method on the CycloneDxLicenseEntry class ?